lib/matplotlib/tests/test_colors.py

```diff
@@ -1728,3 +1728,29 @@ def test_colorizer_vmin_vmax():
     assert ca.vmax == 3.0
     assert ca.norm.vmin == 1.0
     assert ca.norm.vmax == 3.0
+
+
+def test_lsc_from_list_color_alpha_tuple():
+    """
+    GitHub issue #29042: A bug in 'from_list' causes an error
+    when passing a tuple (str, float) where the string is a
+    color name or grayscale value and float is an alpha value.
+    """
+    valid_colors = [("red", 0.3), ("0.42", 0.1), "green"]
+    mcolors.LinearSegmentedColormap.from_list("testcolormap", valid_colors)
+
+
+@pytest.mark.parametrize("colors, error",
+                         [([(0.42, "blue"), (.1, .1, .1, .1)], ValueError),
+                          (["blue", (0.42, "red")], ValueError),
+                          (["blue", (.1, .1, .1, .1), ("red", 2)], ValueError),
+                          ([(0, "red"), (1.1, "blue")], ValueError),
+                          ([(0.52, "red"), (0.42, "blue")], ValueError)])
+def test_lsc_from_list_invalid_inputs(colors, error):
+    with pytest.raises(error):
```

Comment:
Since all raise ValueError, let's keep it simple:

```suggestion
@pytest.mark.parametrize("colors",
                         [[(0.42, "blue"), (.1, .1, .1, .1)],
                          ["blue", (0.42, "red")],
                          ["blue", (.1, .1, .1, .1), ("red", 2)],
                          [(0, "red"), (1.1, "blue")],
                          [(0.52, "red"), (0.42, "blue")],
def test_lsc_from_list_invalid_inputs(colors):
    with pytest.raises(ValueError):
```
*If* you want to be more specific (and better document what the parameters actually test), I suggest to parametrize the error message (example made up - the message is likely different):
```
@pytest.mark.parametrize("colors, msg",
                         ([[(0.42, "blue"), (.1, .1, .1, .1)], "alpha must be between 0 and 1"
                         ...
def test_lsc_from_list_invalid_inputs(colors, msg):
    with pytest.raises(ValueError, match=msg):
```

Reply:
Oups, these error types are still a remnant of the previous version where different error types were raised. I will remove them, thanks for pointing this out!

I don't think matching the error message is of any particular importance – actually, neither is the error type itself. The error don't originate from `from_list` directly, so as long as any underlying function raises any error, we should be ok here.